From e225c3e78a6312d41cc4810190ecb7804ec718b6 Mon Sep 17 00:00:00 2001
Message-ID: <e225c3e78a6312d41cc4810190ecb7804ec718b6.1701297273.git.sharlatanus@gmail.com>
In-Reply-To: <123a0b9b76c24c29184c527dabb652384f7be762.1701297273.git.sharlatanus@gmail.com>
References: <123a0b9b76c24c29184c527dabb652384f7be762.1701297273.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 29 Oct 2023 23:11:05 +0000
Subject: [PATCH v3 61/66] gnu: go-github-com-stretchr-objx: Move to (gnu
 packages golang-web).

* gnu/packages/golang.scm (go-github-com-stretchr-objx,
go-github-com-stretchr-testify-bootstrap): Move from here...
* gnu/packages/golang-web.scm: ...to here.

Change-Id: I357edfb72b789da67f821ae63afe30f18fc1cb39
---
 gnu/packages/golang-check.scm | 51 ++++++++++++++++++++++++++++++++++
 gnu/packages/golang.scm       | 52 -----------------------------------
 2 files changed, 51 insertions(+), 52 deletions(-)

diff --git a/gnu/packages/golang-check.scm b/gnu/packages/golang-check.scm
index 42e501a6e3..b74864d808 100644
--- a/gnu/packages/golang-check.scm
+++ b/gnu/packages/golang-check.scm
@@ -378,6 +378,46 @@ (define-public go-github-com-onsi-ginkgo
 Gomega matcher library.")
     (license license:expat)))
 
+(define-public go-github-com-stretchr-objx
+  (package
+    (name "go-github-com-stretchr-objx")
+    (version "0.4.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/stretchr/objx")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0dygds32qxx6x1x2mmn7msyjr15qi5r70pyzv8dz8cprxq32nzc1"))
+        (modules '((guix build utils)))
+        (snippet
+         '(begin
+            (delete-file-recursively "vendor")))))
+    (build-system go-build-system)
+    (arguments
+     (list
+       #:import-path "github.com/stretchr/objx"
+       #:phases
+       #~(modify-phases %standard-phases
+           (replace 'check
+             (lambda* (#:key inputs #:allow-other-keys #:rest args)
+               (unless
+                 ;; The tests fail when run with gccgo.
+                 (false-if-exception (search-input-file inputs "/bin/gccgo"))
+                 (apply (assoc-ref %standard-phases 'check) args)))))))
+    (propagated-inputs
+     (list go-github-com-davecgh-go-spew
+           go-github-com-pmezard-go-difflib))
+    (inputs
+     (list go-github-com-stretchr-testify-bootstrap))
+    (home-page "https://github.com/stretchr/objx")
+    (synopsis "Go package for dealing with maps, slices, JSON and other data")
+    (description "This package provides a Go library for dealing with maps,
+slices, JSON and other data.")
+    (license license:expat)))
+
 (define-public go-github-com-stretchr-testify
   (package
     (name "go-github-com-stretchr-testify")
@@ -414,6 +454,17 @@ (define-public go-github-com-stretchr-testify
 @end itemize")
     (license license:expat)))
 
+(define go-github-com-stretchr-testify-bootstrap
+  (package
+    (inherit go-github-com-stretchr-testify)
+    (arguments
+     '(#:import-path "github.com/stretchr/testify"
+       #:tests? #f
+       #:phases (modify-phases %standard-phases
+                  (delete 'build))))
+    (propagated-inputs
+     (list go-gopkg-in-yaml-v3))))
+
 (define-public go-github-com-tdewolff-test
   (package
     (name "go-github-com-tdewolff-test")
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 1fb656c547..1870eadb9c 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -4146,58 +4146,6 @@ (define-public go-github-com-jpillora-backoff
 Go.")
       (license license:expat))))
 
-(define-public go-github-com-stretchr-objx
-  (package
-    (name "go-github-com-stretchr-objx")
-    (version "0.4.0")
-    (source
-     (origin
-       (method git-fetch)
-       (uri (git-reference
-             (url "https://github.com/stretchr/objx")
-             (commit (string-append "v" version))))
-       (file-name (git-file-name name version))
-       (sha256
-        (base32
-         "0dygds32qxx6x1x2mmn7msyjr15qi5r70pyzv8dz8cprxq32nzc1"))
-        (modules '((guix build utils)))
-        (snippet
-         '(begin
-            (delete-file-recursively "vendor")))))
-    (build-system go-build-system)
-    (arguments
-     (list
-       #:import-path "github.com/stretchr/objx"
-       #:phases
-       #~(modify-phases %standard-phases
-           (replace 'check
-             (lambda* (#:key inputs #:allow-other-keys #:rest args)
-               (unless
-                 ;; The tests fail when run with gccgo.
-                 (false-if-exception (search-input-file inputs "/bin/gccgo"))
-                 (apply (assoc-ref %standard-phases 'check) args)))))))
-    (propagated-inputs
-     (list go-github-com-davecgh-go-spew
-           go-github-com-pmezard-go-difflib))
-    (inputs
-     (list go-github-com-stretchr-testify-bootstrap))
-    (home-page "https://github.com/stretchr/objx")
-    (synopsis "Go package for dealing with maps, slices, JSON and other data")
-    (description "This package provides a Go library for dealing with maps,
-slices, JSON and other data.")
-    (license license:expat)))
-
-(define go-github-com-stretchr-testify-bootstrap
-  (package
-    (inherit go-github-com-stretchr-testify)
-    (arguments
-     '(#:import-path "github.com/stretchr/testify"
-       #:tests? #f
-       #:phases (modify-phases %standard-phases
-                  (delete 'build))))
-    (propagated-inputs
-     (list go-gopkg-in-yaml-v3))))
-
 (define-public go-github-com-technoweenie-multipartstreamer
   (package
     (name "go-github-com-technoweenie-multipartstreamer")
-- 
2.41.0

